integration_tests/testsuite/test_custom.py

```diff
@@ -39,19 +57,175 @@ def runTest(self):
         if not output:
             logging.debug('No custom tests specified.')
         else:
-            for test_info in json.loads(output):
+            for specification in json.loads(output):
                 test_num += 1
-                name = test_info.get('name', 'test_{0}'.format(test_num))
-                path = test_info.get('path')
-                if path is None:
-                    logging.warn('Test \'%s\' has no path specified! '
-                                 'Skipping...', name)
-                    continue
+                self._run_test_for_specification(specification, test_num)
+
+    def _run_test_for_specification(self, specification, test_num):
+        """Given the specification for a test execute the steps and
+            validate the result.
+
+        Args:
+            specification: Dictionary containing the specification.
+            test_num: Identifier of the test.
+
+        Returns:
+            None.
+        """
+        name = specification.get('name', 'test_{0}'.format(test_num))
+        timeout = specification.get('timeout', 500)
+        steps = specification.get('steps')
+        validation = specification.get('validation')
```

Comment:
if no validation is provided here, `None` will be passed to the call to `_validate()`, which will choke. I think you want `specification.get('validation', [])` here

Reply:
Indeed and `steps` too, if null, could cause a failure due to the for loop.
I think that a good approach would be to fail explicitly if `validation` is not present, as a test without assertion does not make sense, and to add a default value to `steps` as it could be optional.